src/azure-cli/azure/cli/command_modules/acs/_params.py

```diff
@@ -388,10 +388,14 @@ def load_arguments(self, _):
         c.argument('acr', validator=validate_registry_name)
         c.argument('node_name')
 
-    with self.argument_context('aks nodepool', resource_type=ResourceType.MGMT_CONTAINERSERVICE, operation_group='managed_clusters') as c:
+    with self.argument_context('aks nodepool', resource_type=ResourceType.MGMT_CONTAINERSERVICE, operation_group='agent_pools') as c:
         c.argument('cluster_name', help='The cluster name.')
-        # the following argument is declared for the wait command
-        c.argument('agent_pool_name', options_list=['--nodepool-name', '--agent-pool-name'], validator=validate_agent_pool_name, help='The node pool name.')
+        c.argument('nodepool_name', options_list=['--nodepool-name', '--name', '-n'], validator=validate_nodepool_name, help='The node pool name.')
+
+    with self.argument_context('aks nodepool wait', resource_type=ResourceType.MGMT_CONTAINERSERVICE, operation_group='agent_pools') as c:
+        c.argument('resource_name', options_list=['--cluster-name'], help='The cluster name.')
+        # the option name '--agent-pool-name' is depracated, left for compatibility only
+        c.argument('agent_pool_name', options_list=['--nodepool-name', '--name', '-n', '--agent-pool-name'], validator=validate_agent_pool_name, help='The node pool name.')
```

Comment:
It is recommended to split the `--agent-pool-name` which needs to be deprecated and put it into a new hidden argument

Reply:
I think I found an easier way to achieve the above requirements. I'm using the deprecate method that comes with CLI, which will show a warning when this option is used, and it is hidden by default in the help message.

![image](https://user-images.githubusercontent.com/81607949/204217781-4595daab-5138-4107-970d-f5edd75164ed.png)